ZelBack/src/services/explorerService.js

```diff
@@ -346,29 +346,30 @@ async function processInsight(blockDataVerbose, database) {
       }
     }
   }
+}
+
+/**
+ * To process transactions inserts on database and calling app messages.
+ * @param {string} database Database.
+ */
+async function processTransactions(database) {
+  log.info(`processTransactions - Processing ${appsTransactions.length} transactions`);
   if (appsTransactions.length > 0) {
     const options = {
       ordered: false, // If false, continue with remaining inserts when one fails.
     };
     await dbHelper.insertManyToDatabase(database, appsHashesCollection, appsTransactions, options);
-    if (blockDataVerbose.height >= config.fluxapps.fluxAppRequestV2) {
-      while (appsTransactions.length > 500) {
-        appsService.checkAndRequestMultipleApps(appsTransactions.splice(0, 500));
-        // eslint-disable-next-line no-await-in-loop
-        await serviceHelper.delay((5 + (Math.random() * 5)) * 1000); // delay random from 5 to up 10 seconds
-      }
-      if (appsTransactions.length > 0) {
-        appsService.checkAndRequestMultipleApps(appsTransactions);
-      }
-    } else {
-      // eslint-disable-next-line no-restricted-syntax
-      for (const tx of appsTransactions) {
-        appsService.checkAndRequestApp(tx.hash, tx.txid, tx.height, tx.value);
-      }
+    while (appsTransactions.length > 500) {
```

Comment:
I'm missing something here. From what I can tell, we are inserting the data into the database... but then in checkAndRequestMultipleApps we request the appMessage via our peers with the requestAppsMessage before we then call checkAndRequestApp... shouldn't this be the other way around?

Reply:
I'm sure this is just my lack of understanding - I'm not seeing where it checks the permanent app messages that we got via `syncAppsMessages` before it goes off to check the messages from peers